platform/broadcom/sonic-platform-modules-inventec/d7264q28b/modules/inv_swps.h

```diff
@@ -0,0 +1,184 @@
+#ifndef INV_SWPS_H
+#define INV_SWPS_H
+
+#include "transceiver.h"
+#include "io_expander.h"
+
+/* Module settings */
+#define SWP_CLS_NAME          "swps"
+#define SWP_DEV_PORT          "port"
+#define SWP_DEV_MODCTL        "module"
+#define SWP_RESET_PWD         "inventec"
+#define SWP_POLLING_PERIOD    (300)  /* msec */
+#define SWP_POLLING_ENABLE    (1)
+#define SWP_AUTOCONFIG_ENABLE (1)
+
+/* Module information */
+#define SWP_AUTHOR            "Neil <liao.neil@inventec.com>"
+#define SWP_DESC              "Inventec port and transceiver driver"
```

Comment:
transceiver functionality is provided by sfputil, sff8436 driver, why have your own transceiver driver?

Reply:
Guohan
Our implementation has more functionality such as error recovery and error handling on sfp devices. So we prefer to contribute this to use in SONiC.